### ISSUE TITLE
How to post request as binary data

### ISSUE THREAD

**Issue by johnzeng** [parnurzeal/gorequest#112]
Like this curl request:

```shell
curl -X POST \
  -H "Content-Type: image/png" \
  --data-binary '@test.png'  \
  https://xxxx.xxx.xxx/xxxxxx.png
```

I can use http.Post like this:

```go
		httpRequest, err := http.Post(url, "application/json", file)
```

any idea that I can do it with gorequest?

**Comment by fraenky8** [comment]
Maybe this does the trick:

```go
	gorequest.
		New().
		Post("http://httpbin.org/post").
		Type("multipart").
		SendFile("./test.png").
		Set("Content-Type", "image/png").
		End()
```

In you http.Post example, i assume that the content type `application/json` should also be 'image/png'?

**Comment by johnzeng** [comment]
@fraenky8 The reason why I open this issue is because this doesn't work.

**Comment by fraenky8** [comment]
I think i see the problem: The content type `image/png` isn't supported native, therefore i used the `multipart` type. So the `curl` request differs from the `gorequest`-Request, but in the response of `http://httpbin.org/post` it looks like the same result. But multipart adds the boundary-string and this is treated as binary data belonging to the image.

**TL;DR** The content type `image/png` with binary data isn't supported native yet.

**Comment by forrestsun** [comment]
curl --data-binary @filename  -X PUT 
'http://localhost:9000/media/2017/07/21/3db141074052b9be801a2d814ec080b5/3a0c5ec940aa7e5380a78f506209873f.xlsx?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=LNSGUC9EM0KPRD8O174E%2F20170721%2Flocal%2Fs3%2Faws4_request&X-Amz-Date=20170721T150352Z&X-Amz-Expires=86400&X-Amz-SignedHeaders=host&X-Amz-Signature=2aea90e29cbf3b74f0c9147810967e5fc8f9ac82ad681689be3566b7ebe6aa19'

File Upload Ok & md5sum : 5f849c8e56eb61271ae248896a283506

When i use gorequest:
```

	resp, bytebody, errs := req.Put(uplad.URL).
		SetCurlCommand(true).
		Set("Content-Type", "application/octet-stream").
		Type("multipart").
		SendFile(path).End()
```
or 
```
	resp, bytebody, errs := req.Put(uplad.URL).
		Type("multipart").
		SendFile(path).End()
```
File Upload Ok but md5sum:f69f963be299201903163c2cb2213ce1 and file can not open

**Comment by clems71** [comment]
It'd be great to allow one to "send" a io.Reader instance so that it's generic enough : `req.SendReader`.

What do you think?
